src/Filters/coefficients.jl

```diff
@@ -322,9 +322,6 @@ function groupzp(z, p)
         groupedz[i] = splice!(z, closest_zero_idx)
         if !isreal(groupedz[i])
             i += 1
-            if closest_zero_idx > length(z)
-                closest_zero_idx = length(z)
-            end
```

Comment:
This reverts the fix from #433, but the accompanying test still passes, corroborating that this new approach also fixes #432.